cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -666,12 +669,37 @@ public LoadBalancer getLoadBalancer(String resourceGroupName, String loadBalance
         return handleAuthException(() -> azure.loadBalancers().getByResourceGroup(resourceGroupName, loadBalancerName));
     }
 
+    /**
+     * Returns the IP addresses associated with a particular Load Balancer in a particular Azure Resource Group.
+     *
+     * @param resourceGroupName the name of the resource group containing the load balancer
+     * @param loadBalancerName the name of the load balancer
+     * @return IP addresses
+     */
     public List<String> getLoadBalancerIps(String resourceGroupName, String loadBalancerName) {
+        // todo: test this
```

Comment:
Hopefully you did that.

Reply:
Good point. I looked more closely at testing this, but there aren't any existing tests in the codebase for the `AzureClient`.

This method does 3 things:
* retrieve load balancer Public IP address ids
* retrieve Public IP address objects in the resource group
* Adds the CIDR string of the `PublicIpAddress` object to the list of Strings to return

I think that the `AzureClient` should have those three things removed from it, and it should instead live in the `AzureMetadataCollector` class.
* Keeps the AzureClient as just an integration point -- easier to mock, better separation between Azure SDK and business logic
* It's a bit of a pain to move that around.

I've got some other refactoring -- Java doc comments and moving some methods around, sitting in another branch.

I'll make these changes separately and then take a look at opening a different PR. Would like to get _this_ PR merged in.